scripts/west_commands/runners/spi_burn.py

```diff
@@ -0,0 +1,100 @@
+# Copyright (c) 2021, Telink Semiconductor
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+import time
+import subprocess
+
+from runners.core import ZephyrBinaryRunner, RunnerCaps
+
+class SpiBurnBinaryRunner(ZephyrBinaryRunner):
+    '''Runner front-end for SPI_burn.'''
+
+    def __init__(self, cfg, addr, eraseall, spiburn, iceman):
+        super().__init__(cfg)
+
+        self.bin = cfg.bin_file
+        self.spiburn = spiburn
+        self.iceman = iceman
+        self.addr = addr
+        if eraseall:
+            self.eraseall = "--erase-all"
+        else:
+            self.eraseall = ''
+
+    @classmethod
+    def name(cls):
+        return 'spi_burn'
+
+    @classmethod
+    def capabilities(cls):
+        return RunnerCaps(commands={'flash'})
+
+    @classmethod
+    def do_add_parser(cls, parser):
+        parser.add_argument('--addr', default='0x0',
+                            help='start flash address to write')
+        parser.add_argument('--erase-all', default=False, action='store_true',
+                            help='erase flash before burn')
```

Comment:
Hmm, I'm sorry for asking for yet another change, but I somehow missed this argument. This is replicating standard runner functionality, but it's using the wrong argument name. This should be using the standard `--erase` argument instead. Support this by adding `erase=True` to the arguments to `RunnerCaps` above -- see other runners that do the same thing for examples.

Reply:
@mbolivar-nordic, no problem. Requested changes are done.